### PR TITLE
[SPARK-44264][PYTHON] Added Deepspeed Folder to setup.py

### PR DESCRIPTION
**Pull request by mathewjacob1002**
### What changes were proposed in this pull request?
Added pyspark.ml.deepspeed to the setup.py

### Why are the changes needed?
It allows the deepspeed distributor to be pip installed when you pip install pyspark.

### Does this PR introduce _any_ user-facing change?
No

### CODE REVIEW

**Comment by mathewjacob1002** [comment]
cc @HyukjinKwon just adds deepspeed functionalities to the setup.py for pip.

File: python/setup.py [hunk 1]
```diff
@@ -241,6 +241,7 @@ def run(self):
             "pyspark.ml.linalg",
             "pyspark.ml.param",
             "pyspark.ml.torch",
+            "pyspark.ml.deepspeed",
```
Comment (HyukjinKwon) [inline python/setup.py:244]: I think you should also put `__init__.py` at `pyspark/ml/deepspeed` directory to make it as a canonical module.

**Comment by HyukjinKwon** [comment]
Let's also file a JIRA. The fix is small but the impact of this change is actually pretty big.

**Comment by maddiedawson** [comment]
@HyukjinKwon can we reuse this ticket: https://issues.apache.org/jira/browse/SPARK-44264

**Comment by HyukjinKwon** [comment]
Let's add `__init__.py` file, see https://github.com/apache/spark/pull/42123#discussion_r1271872519

**Review by maddiedawson — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.

**Pull request closed without merging**